### PR TITLE
When completing color keys in settings, fill in current value

### LINKED ISSUE #25633: [theming] when completing color keys in settings, fill in current value
That would be very helpful 👍

### PR DESCRIPTION
**Pull request by SimonSiefke**
fixes #25633

color-theme

### CODE REVIEW

File: src/vs/platform/theme/common/colorUtils.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { IJSONSchema, IJSONSchemaMap } from 'vs/base/common/jsonSchema';
 import { IJSONContributionRegistry, Extensions as JSONExtensions } from 'vs/platform/jsonschemas/common/jsonContributionRegistry';
 import * as platform from 'vs/platform/registry/common/platform';
 import { IColorTheme } from 'vs/platform/theme/common/themeService';
+import type { ColorThemeData } from 'vs/workbench/services/themes/common/colorThemeData';
```
Comment (aeschli) [inline src/vs/platform/theme/common/colorUtils.ts:14]: platform can not reference `workbench/services`, use `IColorTheme` instead

File: src/vs/platform/theme/common/colorUtils.ts [hunk 2]
```diff
@@ -211,6 +225,10 @@ export function getColorRegistry(): IColorRegistry {
 	return colorRegistry;
 }
 
+export const setTheme = (theme: ColorThemeData) => {
```
Comment (aeschli) [inline src/vs/platform/theme/common/colorUtils.ts:228]: that's not so nice to have this as a global function. Better add it as a function to the `IColorRegistry`. I would call it` notifyThemeUpdate(...)`

File: src/vs/workbench/services/themes/browser/workbenchThemeService.ts [hunk 3]
```diff
@@ -486,6 +486,7 @@ export class WorkbenchThemeService extends Disposable implements IWorkbenchTheme
 	}
 
 	private applyTheme(newTheme: ColorThemeData, settingsTarget: ThemeSettingTarget, silent = false): Promise<IWorkbenchColorTheme | null> {
+		setTheme(newTheme)
```
Comment (aeschli) [inline src/vs/workbench/services/themes/browser/workbenchThemeService.ts:489]: theme colors not only change on `applyTheme`, but also when the color customizations change.
Best is to subscribe to `onColorThemeChange` to get all notifications

**Review by aeschli — CHANGES_REQUESTED** [review]

File: src/vs/workbench/services/themes/browser/workbenchThemeService.ts [hunk 4]
```diff
@@ -136,6 +136,9 @@ export class WorkbenchThemeService extends Disposable implements IWorkbenchTheme
 		this.currentProductIconTheme = ProductIconThemeData.createUnloadedTheme('');
 		this.productIconThemeSequencer = new Sequencer();
 
+		this.onColorThemeChange.event
```
Comment (aeschli) [inline src/vs/workbench/services/themes/browser/workbenchThemeService.ts:139]: this line can be removed

File: src/vs/workbench/services/themes/browser/workbenchThemeService.ts [hunk 5]
```diff
@@ -136,6 +136,9 @@ export class WorkbenchThemeService extends Disposable implements IWorkbenchTheme
...
+		this._register(this.onDidColorThemeChange(this.handleColorThemeChange, this));
```
Comment (aeschli) [inline src/vs/workbench/services/themes/browser/workbenchThemeService.ts:140]: directly call `getColorRegistry().notifyThemeUpdate(theme)`

**Review by aeschli — CHANGES_REQUESTED** [review]

**Review by bpasero — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

**Review by aeschli — APPROVED** [review]

**Comment by aeschli** [comment]
Thanks @SimonSiefke !

**Pull request merged**